### PR TITLE
[T5A1][W13-B4] Conan Kian

### PR DESCRIPTION
**Pull request by ckjr**
Ready for review

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -82,6 +82,32 @@ Use case ends.
 
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
+  
+#### Use case: Edit tag
```
Comment (ckjr) [inline doc/DeveloperGuide.md]: Note: This was done before the change from "Edit Tag" to "Rename Tag". This use case is assuming a edit of a tag for a specific person, not a tag as a whole that applies to all people with that tag.

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -43,10 +43,19 @@ Priority | As a ... | I want to ... | So that I can...
 -------- | :-------- | :--------- | :-----------
 `* * *` | new user | see usage instructions | refer to instructions when I forget how to use the App
 `* * *` | user | add a new person |
+`* * *` | user | edit a person | update the person's details
 `* * *` | user | delete a person | remove entries that I no longer need
 `* * *` | user | find a person by name | locate details of persons without having to go through the entire list
 `* *` | user | hide [private contact details](#private-contact-detail) by default | to minimize chance of someone else seeing them by accident
+`* *` | new user | export and import contacts in a nice to read format |
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: what is "nice" ?
- in general it is better to be as specific and objective as possible.
- words like less, more, fast, etc., are subjective

File: doc/DeveloperGuide.md (hunk 1 above)
Comment (okkhoy) [inline doc/DeveloperGuide.md]: you preempted my comment :-)

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -73,13 +82,48 @@ Use case ends.
...
+  Use case ends
 
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should be able to hold up to 1000 tags per person.
+6. Should be able to save at every change.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: isn't this a functionality?

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -73,13 +82,48 @@ Use case ends.
...
+7. Should be able to handle all possible errors correctly at each input.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: this is also a functionality

File: doc/DeveloperGuide.md [hunk 5]
```diff
@@ -73,13 +82,48 @@ Use case ends.
...
+8. Should be able to give a response to each input within the second.
+9. Should encrypt the addressBook to protect privacy.
+10. Should be able to import and export address books.
+11. Should be able to add functions easily without changing much of the existing code.
+12. Should not take too much memory during runtime.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: how much is too much?
- in general it is better to be as precise and objective as possible.
- words like less, more, fast, etc., are subjective

**Comment by okkhoy** [comment]
@ckjr 
Some comments added. Please ack & close the PR after reading comments.

**Comment by ckjr** [comment]
Acknowledged

**Pull request closed without merging**